### PR TITLE
Avoid race conditions in social user connection creators

### PR DESCRIPTION
**Pull request by leogrim**

### CODE REVIEW

**Comment by leogrim** [comment]
@eishay @raymondng

**Comment by raymondng** [comment]
lgtm. thanks!

File: kifi-backend/modules/shoebox/app/com/keepit/common/social/SocialGraphPluginImpl.scala [hunk 1]
```diff
@@ -64,45 +64,41 @@ private[social] class SocialGraphActor @Inject() (
...
-            val updatedSui = rawInfo.jsons.foldLeft(socialUserInfo)(graph.updateSocialUserInfo)
-            val latestUserValues = rawInfo.jsons.map(graph.extractUserValues).reduce(_ ++ _)
-            db.readWrite { implicit c =>
-              latestUserValues.collect { case (key, value) if userValueRepo.getValue(userId, key) != Some(value) =>
-                userValueRepo.setValue(userId, key, value)
-                heimdal.setUserProperties(userId, key -> ContextStringData(value))
-              }
-              socialRepo.save(updatedSui.withState(SocialUserInfoStates.FETCHED_USING_SELF).withLastGraphRefresh())
+          val updatedSui = rawInfo.jsons.foldLeft(socialUserInfo)(graph.updateSocialUserInfo)
+          val latestUserValues = rawInfo.jsons.map(graph.extractUserValues).reduce(_ ++ _)
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/SocialGraphPluginImpl.scala:85]: Is this reduce(_ ++ _) same as flatten? or just flatMap?

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/SocialGraphPluginImpl.scala:85]: They're Maps, not sure if flatten works actually but I wanted to make explicit the order of key overrides.

Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/SocialGraphPluginImpl.scala:85]: Cool.

**Pull request merged**